beamer/deploy/config.py

```diff
@@ -0,0 +1,81 @@
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import apischema
+from apischema import ValidationError, schema
+from apischema.metadata import validators
+from eth_typing import ChecksumAddress
+from eth_utils import is_checksum_address
+
+from beamer.typing import URL, ChainId
+
+
+def _validate_token_address(address: str) -> None:
+    if address != "mintable_token" and not is_checksum_address(address):
+        raise ValidationError(f"expected either 'mintable_token' or a checksum address: {address}")
+
+
+@dataclass
+class _BaseChain:
+    name: str
+    rpc: str
+    chain_id: ChainId = field(metadata=schema(min=1))
+
+
+@dataclass
+class _RequestManagerArgs:
+    claim_stake: float = field(metadata=schema(min=0))
+    claim_request_extension: int = field(metadata=schema(min=0))
+    claim_period: int = field(metadata=schema(min=0))
+    challenge_period_extension: int = field(metadata=schema(min=0))
+
+
+@dataclass
+class _Fees:
+    min_fee_ppm: int = field(metadata=schema(min=0))
+    lp_fee_ppm: int = field(metadata=schema(min=0))
+    protocol_fee_ppm: int = field(metadata=schema(min=0))
+
+
+@dataclass
+class _Token:
+    token_address: str | ChecksumAddress = field(metadata=validators(_validate_token_address))
+    transfer_limit: int = field(metadata=schema(min=0))
+    eth_in_token: float = field(metadata=schema(min=0))
+
+
+@dataclass
+class Chain:
+    name: str
+    chain_id: ChainId = field(metadata=schema(min=1))
```

Comment:
we can reuse your ChainIdParam check from the next commit, to make sure, wherever we write a chain id, it's gonna be a positive integer

Reply:
You mean, when serializing? But we don't do serialization of config data. This is merely input to us.
On deserialization, the check happens automatically, via apischema.

Reply:
does it also check for positive numbers?

Reply:
ah it does with `field(metadata=schema(min=1))`

Reply:
> ah it does with `field(metadata=schema(min=1))`

Yes, precisely.